lib/src/converter/mod.rs

```diff
@@ -107,6 +107,11 @@ impl<'a, T: Turtle> visit::XmlVisitor for ConversionVisitor<'a, T> {
             warn!("Clip paths are not supported: {:?}", node);
         }
 
+        if node.attribute("style") == Some("display:none") {
+            // Skip any nodes marked as hidden
+            continue;
```

Comment:
It's a bit confusing the way I implemented it, but each node needs to go through the end of the visit method -- there is a transform-popping step that has to run if this is the last child element. So we cannot return early here.

Instead I recommend modifying:
```rust
if node.tag_name().name() == PATH_TAG_NAME
````
to include a check for display:none

Reply:
Whoops!

When I merged from master, I didn't realize that between when I had forked the repo and created the pull request, you had made some pretty drastic changes to the way you were navigating through the nodes. I wrote this if statement when you still had a while loop to do it.

I took a new look at the current implementation and moved the place I'm checking for the "display:none" to where i _think_ it needs to go. I put it as a check in the depth_first_visit function. I think it has to go here because it needs to catch things like g elements and prevent the search from looking through the children of those elements. 

That being said this is my first time working with rust and hope I'm not breaking any patterns. Let me know if this should go somewhere else or if I'm missing some obvious cleaner way to do it or something. Don't wanna add ugly code to your nice repo.

Oh also I changed the check to look and see if the style _contains_ "display:none" rather than is exactly equal to it. Could catch a few cases where an element has a style that is hidden but also has other stuff in it.

Reply:
Thanks for revising, this looks good. I think the transform problem still applies though. Consider the scenario:
```
<g transform="scale(1 0.5)" >
  <p style="display:none" />
</g>
```

We visit g first and push the scale transform onto the stack. We do not visit g again and must rely on visiting p to pop that transform.

Reply:
Maybe the solution there is to split visiting out into visit_begin and visit_end methods. What do you think?

Reply:
Ahh I see yeah.

Hmm. Does having 2 methods solve that? Since the search is being done by just pushing the children onto the stack and then continuing the while loop, I don't know where you'd put the beginning and end so they'd wrap all the stuff they need to successfully. When I do depth first stuff like this I usually go the recursion route, which makes it possible to do the begin/end stuff you're talking about, and I'm not super familiar with this stack-based approach.

One other option I just thought of is we could maybe just trim the input of all the display:none nodes before even starting the depth_first_visit. I'm not sure if the object is mutable or how that would work in rust but if that works, it gets around having to deal with any of the transforms stuff. 

I'm not sure the best way to deal with it. If you've got an idea that you think will work go for it! Or I can also take a stab at one of these options if that works better for ya.

Reply:
The recursion route is probably the easier way to go. I'll merge this and revise to account for that edge case.